tailcfg/tailcfg.go

```diff
@@ -574,9 +584,47 @@ func (req *RegisterRequest) Clone() *RegisterRequest {
 		tok := *res.Auth.Oauth2Token
 		res.Auth.Oauth2Token = &tok
 	}
+	if res.DeviceCert != nil {
+		res.DeviceCert = make([]byte, len(req.DeviceCert))
+		copy(res.DeviceCert, req.DeviceCert)
+	}
+	if res.Signature != nil {
+		res.Signature = make([]byte, len(req.Signature))
+		copy(res.Signature, req.Signature)
+	}
 	return res
 }
 
+// WritePlaintextForSigning writes the signed fields (along with the server and machine
+// public keys) to w needed to sign or verify a RegsterRequest.
+func (req *RegisterRequest) WritePlaintextForSigning(
```

Comment:
Typically, this package doesn't have any code of note in it. It serves as our equivalent of a .proto file for sharing between the server and the client. (With the list of unavoidable collection of exceptions you might expect, around this diff.) Is there somewhere else this code could live?

Reply:
This code is needed on both the server and the client to ensure that they generate the same plaintext, so this seemed like the most appropriate place. This is the minimal code I could think of sharing here, but another option would be for it to return the sha256 from this receiver, which maybe would make it act like more like a generated field rather than exposing internal implementation details?

Reply:
... or if you have another suggestion for where to put a bit of shared code between the client and server, I'm happy to hear about it.

Reply:
The x509 dep in particular seems a bit heavy for this package.

Reply:
Hmm, x509 is just needed for x509.SignatureAlgorithm, which I only included for future flexibility. Maybe I should just make a new enum with one value for now, and then let the other code convert that to the appropriate x509 type as needed.

Reply:
New type SGTM.

Reply:
Does this look better?